types/test/typescriptDocs/ModelInit.ts

```diff
@@ -204,6 +201,7 @@ async function doStuffWithUser() {
 
   const project = await newUser.createProject({
     name: "first!",
+    ownerId: 123,
```

Comment:
we are investigating some issues in our app after upgrading to Sequelize 6.12.0 that might be related to this PR.

Since user-> project association is defined like this:

```ts
User.hasMany(Project, { sourceKey: "id", foreignKey: "ownerId", as: "projects" });
```

Shouldn't `ownerId` be automatically set when you do `await newUser.createProject({ name: "first!" })`?

Looking at the code in [lib/associations/has-many.js](https://github.com/sequelize/sequelize/blob/main/lib/associations/has-many.js#L481), it seems like the `ownerId` will even be overwritten with `newUser.get('id')`.

I will try to open a PR with an idea to fix it, but wanted to first confirm this is not expected behavior

Reply:
@EmilioCristalli issue being that you need to specify the id otherwise it fails TS?

Reply:
yes.
Since `ownerId` is going to be automatically set, i think it shouldn't be required by TS. Or even not allowed, since it will be overwritten

Reply:
yeah, that's a good catch

Reply:
i just opened this PR https://github.com/sequelize/sequelize/pull/13892 with an idea to fix it